data/lib/core/creature.lua

```diff
@@ -178,3 +178,29 @@ end
 function Creature.getMonster(self)
 	return self:isMonster() and self or nil
 end
+
+function Creature.getKillers(self)
+	local killers = {}
+	local inFightTicks = configManager.getNumber(configKeys.PZ_LOCKED)
+	local timeNow = os.mtime()
+	for cid, cb in pairs(self:getDamageMap()) do
+		local creature = Creature(cid)
+		if creature and creature ~= self and (timeNow - cb.ticks) <= inFightTicks then
+			killers[#killers + 1] = creature
+		end
+	end
+	return killers
+end
```

Comment:
What if this function and the one below are merged? Return a table of killers sorted by most damage first. It doesn't add to the code complexity though, so if it the code ends up ugly, just dismiss that 😅